src/liballoc/collections/btree/node.rs

```diff
@@ -370,23 +370,32 @@ impl<BorrowType, K, V, Type> NodeRef<BorrowType, K, V, Type> {
         NodeRef { height: self.height, node: self.node, root: self.root, _marker: PhantomData }
     }
 
-    /// Assert that this is indeed a proper leaf node, and not the shared root.
+    /// The caller must ensure that this is a proper node, and not the shared root.
```

Comment:
Nit: since "proper node" is the antonym of "shared root", I would prefer
```suggestion
    /// The caller must ensure that this is a proper node, not the shared root.
```
As-is, it sounds a little too much like a conjunction IMO: "this must be a proper node, and furthermore it must not be the shared root" vs the intended "this must be a proper node (reminder: proper node means any node other than the shared root)".

(This applies to all copies of this phrase.)

Reply:
I just tried to recycle as much as possible from the original. There is no definition of proper node elsewhere, or any kind of properness, so why not just say "The caller must ensure that this(*) is not the shared root"?

(*) or "this node" or "self" or "we are" or ...

Reply:
Oh yeah, that's probably better.